amazon-redshift-plugin/src/main/java/io/cdap/plugin/amazon/redshift/RedshiftSchemaReader.java

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.amazon.redshift;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.db.CommonSchemaReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.List;
+import java.util.Set;
+
+/**
+ *    Redshift Schema Reader class
+ */
+public class RedshiftSchemaReader extends CommonSchemaReader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RedshiftSchemaReader.class);
+
+  public static final Set<Integer> STRING_MAPPED_REDSHIFT_TYPES = ImmutableSet.of(
+    Types.OTHER, Types.ARRAY, Types.SQLXML
+  );
+
+  public static final Set<String> STRING_MAPPED_REDSHIFT_TYPES_NAMES = ImmutableSet.of(
+    "timetz", "money" 
+  );
+
+  private final String sessionID;
+
+  public RedshiftSchemaReader() {
+    this(null);
+  }
+
+  public RedshiftSchemaReader(String sessionID) {
+    super();
+    this.sessionID = sessionID;
+  }
+
+  @Override
+  public Schema getSchema(ResultSetMetaData metadata, int index) throws SQLException {
+    String typeName = metadata.getColumnTypeName(index);
+    int columnType = metadata.getColumnType(index);
+
+    if (STRING_MAPPED_REDSHIFT_TYPES.contains(columnType) || STRING_MAPPED_REDSHIFT_TYPES_NAMES.contains(typeName)) {
+      return Schema.of(Schema.Type.STRING);
+    }
+    if (typeName.equalsIgnoreCase("INT")) {
+      return Schema.of(Schema.Type.INT);
+    }
+    if (typeName.equalsIgnoreCase("BIGINT")) {
+      return Schema.of(Schema.Type.LONG);
+    }
+
+    // If it is a numeric type without precision then use the Schema of String to avoid any precision loss
+    if (Types.NUMERIC == columnType) {
+      int precision = metadata.getPrecision(index);
+      if (precision == 0) {
+        LOG.warn(String.format("Field '%s' is a %s type without precision and scale, "
+                                 + "converting into STRING type to avoid any precision loss.",
+                               metadata.getColumnName(index),
+                               metadata.getColumnTypeName(index)));
+        return Schema.of(Schema.Type.STRING);
+      }
+    }
+
+    if (typeName.equalsIgnoreCase("timestamp")) {
+      return Schema.of(Schema.LogicalType.DATETIME);
+    }
+
+    return super.getSchema(metadata, index);
+  }
+
+  @Override
+  public boolean shouldIgnoreColumn(ResultSetMetaData metadata, int index) throws SQLException {
+    if (sessionID == null) {
+      return false;
+    }
+    return metadata.getColumnName(index).equals("c_" + sessionID) ||
```

Comment:
What exactly are these columns?

Reply:
This is available in all DB plugins with connector. I have tried to find out its usage, not much documentation is available. So I put it as it is.

Reply:
@vikasrathee-cs The PR is approved. Can this be merged now?

Reply:
yes Saurabh, merging it.